src/modifiers/hide.js

```
@@ -0,0 +1,47 @@
+// @flow
+import type { ModifierArguments, Modifier, Options } from '../types';
+import { top, bottom, left, right } from '../enums';
+
+type ModifierData = {
+  top: number,
+  bottom: number,
+  right: number,
+  left: number,
+};
+
+const isAnySideFullyClipped = (overflow: ModifierData): boolean =>
+  [top, right, bottom, left].some(side => overflow[side] >= 0);
+
+export function hide({ state }: ModifierArguments<Options>) {
+  const reference = state.measures.reference;
+  const overflow = state.modifiersData.detectOverflowHide.clippingArea;
```

I would probably make `detectOverflowHide` a configurable option.

```
  const overflow = state.modifiersData[options.detectOverflowModifierName].clippingArea;
```

---

How does that provide any benefit?

---

Users can recycle the same detectOverflow modifier for various modifiers, rather than having to define a new one each time.

---

A problem with that is how do we specify `requires`?

---

For now I’d ask users to make sure they set the proper dependency names, we could build some fancy solution but it’s overkill for now

---

For now I’d ask users to make sure they set the proper dependency names, we could build some fancy solution but it’s overkill for now

---

It makes `detectOverflowName` option pointless however?

---

Why? We can't read from the `requires` list to know which one to use, it may have a completely different name we can't recognize.

---

The problem is the user needs to spread in the original `requires` as well into their custom specified `requires` property, and it will contain the default name for hide's `detectOverflow`?

---

I misread your `can't` as `can`, sorry 🤦‍♂.

I see what you're saying.